ivadomed/utils.py

```diff
@@ -704,11 +435,9 @@ def segment_volume(folder_model, fname_image, fname_roi=None):
 
         # Reconstruct 3D object
         for i_slice in range(len(preds)):
-            for i_label in range(preds[i_slice].shape[0]):
-                batch["input_metadata"][i_slice][i_label]['data_type'] = 'gt'
             # undo transformations
             preds_i_undo, metadata_idx = undo_transforms(preds[i_slice],
-                                                         batch["input_metadata"][i_slice],
+                                                         batch["gt_metadata"][i_slice],
```

Comment:
For me 'gt_metadata' is empty here (segment volume used for inference), that's why input_metadata was used. The data type of input metadata needs to be modified to 'gt'. With this modification it could be used as gt_metadata.